csrc/device_lower/pass/circular_buffer.cpp

```diff
@@ -334,6 +334,26 @@ class CloneTmaCircularBufferLoopAndInsertSync
         war_mbarriers_to_uses_(getAllWarMbarriersToUses()),
         war_mbarriers_to_wait_(getAllMbarriersToWait()) {}
 
+  bool clonesCircularBufferLoad() const {
+    return loop_type_ == CircularBufferLoopStage::Prolog ||
+        loop_type_ == CircularBufferLoopStage::Main;
+  }
+
+  bool clonesCompute() const {
```

Comment:
```suggestion
  // The compute expressions are cloned for these circular buffer loop types.
  // e.g., No actual computation occurs in the Prologue stage.
  bool clonesCompute() const {
```

Reply:
I moved the logic to `type.h` and added comments there.